### PR TITLE
[static op generation] lstsq

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Description

https://github.com/PaddlePaddle/Paddle/issues/53267 case12

### CODE REVIEW

File: paddle/phi/api/yaml/ops.yaml [hunk 1]
```diff
@@ -1194,6 +1194,15 @@
     func : logsigmoid
   backward : logsigmoid_grad
 
+- op : lstsq
+  args : (Tensor x, Tensor y, Scalar rcond, str driver)
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/ops.yaml]: 增加默认值`Scalar rcond=0.0f, str driver="gels"`

File: paddle/phi/api/yaml/ops.yaml [hunk 2]
```diff
@@ -1194,6 +1194,15 @@
...
+  output : Tensor(solution), Tensor(residuals), Tensor(rank), Tensor(singular_values)
+  infer_meta :
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/ops.yaml:1200]: 增加
```
optional : residuals
```
注意对比`lstsq_op.cc`

File: paddle/phi/api/yaml/op_compat.yaml [hunk 3]
```diff
@@ -1379,6 +1379,14 @@
   extra :
     attrs : [bool use_mkldnn = false, bool is_test = false]
 
+- op : lstsq
+  input :
+    {x : X, y : Y}
+  attrs:
+    {rcond : rcond, driver : driver}
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/op_compat.yaml]: `attrs`名字一样就不需要映射了，注意阅读相关说明

Comment (Liyulingyue) [inline paddle/phi/api/yaml/op_compat.yaml]:
```suggestion
```

File: paddle/phi/api/yaml/ops.yaml (hunk 1 above)
Comment (Liyulingyue) [reply paddle/phi/api/yaml/ops.yaml]:
```suggestion
  args : (Tensor x, Tensor y, Scalar rcond=0.0f, str driver="gels")
```

File: paddle/phi/api/yaml/ops.yaml [hunk 4]
```diff
@@ -1194,6 +1194,15 @@
...
 
+- op : lstsq
+  args : (Tensor x, Tensor y, Scalar rcond=0.0f, str driver="gels")
+  output : Tensor(solution), Tensor(residuals), Tensor(rank), Tensor(singular_values)
+  infer_meta :
+    func : LstsqInferMeta
+  kernel :
+    func : lstsq
+    data_type : x
+
```
Comment (Liyulingyue) [inline paddle/phi/api/yaml/ops.yaml:1206]:
```suggestion
  optional : residuals

```

File: paddle/phi/api/yaml/op_compat.yaml [hunk 5]
```diff
@@ -1379,6 +1379,12 @@
   extra :
     attrs : [bool use_mkldnn = false, bool is_test = false]
 
+- op : lstsq
+  input :
+    {x : X, y : Y}
+  output :
+    {solution : Solution, residuals : Residuals, rank : Rank, singular_values : SingularValues}
```
Comment (zyfncg) [inline paddle/phi/api/yaml/op_compat.yaml]: input -> inputs
output -> outputs

File: paddle/phi/api/yaml/op_compat.yaml [hunk 6]
```diff
@@ -1379,6 +1379,12 @@
   extra :
     attrs : [bool use_mkldnn = false, bool is_test = false]
 
+- op : lstsq
+  input :
```
Comment (Liyulingyue) [inline paddle/phi/api/yaml/op_compat.yaml]:
```suggestion
  inputs :
```

File: paddle/phi/api/yaml/op_compat.yaml [hunk 7]
```diff
@@ -1379,6 +1379,12 @@
   extra :
     attrs : [bool use_mkldnn = false, bool is_test = false]
 
+- op : lstsq
+  inputs :
+    {x : X, y : Y}
+  output :
```
Comment (Liyulingyue) [inline paddle/phi/api/yaml/op_compat.yaml]:
```suggestion
  outputs :
```

File: paddle/phi/api/yaml/op_compat.yaml [hunk 8]
```diff
@@ -1379,6 +1379,12 @@
   extra :
     attrs : [bool use_mkldnn = false, bool is_test = false]
 
+- op : lstsq
+  inputs :
+    {x : X, y : Y}
+  outputs :
+    {solution : Solution, residuals : Residuals, rank : Rank, singular_values : SingularValues}
+
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/op_compat.yaml:1391]: 加上:
```
scalar :
  rcond :
    data_type : float
    support_tensor : true
```

**Review by heavyrain-lzy — APPROVED** [review]
LGTM

**Pull request merged**